tests/per_provider/persistent_after.rs

```diff
@@ -51,6 +52,6 @@ fn should_have_been_deleted() {
         client
             .destroy_key(key_name)
             .expect_err("This key should have been destroyed."),
-        ResponseStatus::PsaErrorDoesNotExist
+        Error::Service(ResponseStatus::PsaErrorDoesNotExist)
```

Comment:
If you want, to avoid having to wrap all `ResponseStatus` in `Error::Service`, you could make the test client return `Result<_, ResponseStatus>` panicking if the error is not `Error::Service`.
I think all of the test client requests should go to the service and back, otherwise we are not really testing the service.

Reply:
I didn't mind it that much, but I can change it to unwrap the value before returning